source/GStreamerMSEMediaPlayerClient.cpp

```diff
@@ -32,20 +32,36 @@ const int64_t segmentStartMaximumDiff = 1000000000;
 
 GStreamerMSEMediaPlayerClient::GStreamerMSEMediaPlayerClient(
     const std::shared_ptr<firebolt::rialto::client::ClientBackendInterface> &ClientBackend)
-    : mClientBackend(ClientBackend), mPosition(0), mDuration(0), mIsConnected(false),
-      mMaxWidth(DEFAULT_MAX_VIDEO_WIDTH), mMaxHeight(DEFAULT_MAX_VIDEO_HEIGHT), mVideoRectangle{0, 0, 1920, 1080}
+    : mClientBackend(ClientBackend), mPosition(0), mDuration(0), mIsConnected(false), 
+      mMaxWidth(DEFAULT_MAX_VIDEO_WIDTH), mMaxHeight(DEFAULT_MAX_VIDEO_HEIGHT), mVideoRectangle{0, 0, 1920, 1080}, mStreamingStopped(false)
 {
     mBackendQueue.start();
 }
 
 GStreamerMSEMediaPlayerClient::~GStreamerMSEMediaPlayerClient()
+{
+    if(!mStreamingStopped)
```

Comment:
Something small, but wouldn't it be better if  if(!mStreamingStopped) check be inside stopStreaming ?

Reply:
done